### DISCUSSION TITLE
Fancy Conversation Switching Animation

### DISCUSSION THREAD

**Discussion by CNSeniorious000** [anse-app/anse#36] [Show and tell]
Would you like some fancy but elegant transition animation? I have spent a lot of effort on the speed curve of text fade-in and fade-out, to ensure a smooth and elastic effect.

https://github.com/anse-app/anse/assets/74518716/bdced721-c9ef-4fe7-8749-27b52ea9b3c4

**Comment by CNSeniorious000** [comment]
P.S. Actually I do add a new icon set into `package.json` and applied `active:scale-95` to buttons. I understand that this may not be in line with the design style of the project, **but they are not necessary.** In fact, just using an icon in the *Carbon* can makes the result fine enough.

I'm thinking, is it a good idea to add some animations into this UI? You know, OpenAI's current UI contains transitions in the conversation list, and other UIs like *Bing* or *Bard*'s also add animations to the dialogue bubbles. On the other hand, there are a number of ChatGPT / other AI mirror sites do so. What do you think?

**Comment by ddiu8081** [comment]
Thanks, this animation looks great!

I'd prefer to only enable the title animation when the AI summarizes the title though, to represent that it was generated by the AI. And keep the title changed immediately when setting the title manually or switching conversations.

Also, while I don't think it's a good idea to play the title animation when switching conversations, I think it's great to do a full page animation. But I plan to do front-end routing in the future, so maybe I can choose to add animations on route switching.

**Comment by CNSeniorious000** [comment]
Yeah, implementing flashy animations without necessity was just a fun experiment. And wow, glad to hear we are getting routing, with which many features will become possible! Looking forward ~
